### PR TITLE
Bump Quarto version -> 1.3.340

### PR DESCRIPTION
**Pull request by msarahan**
This moves things forward a little bit, but isn't working yet. It copies the expected ~2000 files, but something's not quite right yet:

```
ERROR: Theme file compilation failed:

Unexpected exception:
Invalid argument (uri): Value must be a String or a Uri: null


package:path/src/context.dart 1119                                                         _parseUri
package:path/src/context.dart 1040                                                         Context.fromUri
package:path/path.dart 435                                                                 fromUri
/home/conda/feedstock_root/build_artifacts/dart-sass_1678130103876/work/bin/sass.dart 135  _loadVersion

Error: Theme file compilation failed:

Unexpected exception:
Invalid argument (uri): Value must be a String or a Uri: null


package:path/src/context.dart 1119                                                         _parseUri
package:path/src/context.dart 1040                                                         Context.fromUri
package:path/path.dart 435                                                                 fromUri
/home/conda/feedstock_root/build_artifacts/dart-sass_1678130103876/work/bin/sass.dart 135  _loadVersion
    at dartCommand (file:///home/msarahan/mambaforge/conda-bld/quarto_1683570031711/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/bin/quarto.js:59889:15)
    at async checkVersions (file:///home/msarahan/mambaforge/conda-bld/quarto_1683570031711/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/bin/quarto.js:104040:26)
    at async check (file:///home/msarahan/mambaforge/conda-bld/quarto_1683570031711/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/bin/quarto.js:103997:13)
    at async Command.fn (file:///home/msarahan/mambaforge/conda-bld/quarto_1683570031711/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/bin/quarto.js:104220:5)
    at async Command.execute (file:///home/msarahan/mambaforge/conda-bld/quarto_1683570031711/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/bin/quarto.js:8437:13)
    at async quarto (file:///home/msarahan/mambaforge/conda-bld/quarto_1683570031711/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/bin/quarto.js:127548:5)
    at async file:///home/msarahan/mambaforge/conda-bld/quarto_1683570031711/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/bin/quarto.js:127566:9
Tests failed for quarto-1.3.340-0.tar.bz2 - moving package to /home/msarahan/mambaforge/conda-bld/broken
```

The `env` call in meta.yaml in the tests section was to see if all the env vars are set right. They appear to be set correctly. This could be a problem with the dart-sass package.

### CODE REVIEW

**Comment by mfisher87** [comment]
Nice. Reproduced the same error on my end, playing around looking for a fix now.

**Comment by mfisher87** [comment]
Since the runtime dependency on dart-sass isn't pinned, a newer version than the one in `configuration` was being used (`1.58.3` instead of `1.55.0`).

I tried pinning to `1.55.0` but this version isn't available on `conda-forge`, so I tried the next-closest (`v1.54.4`), and that yields the same error as `1.58.3`. Looks like I need to dig deeper than just trying different versions :)

**Comment by mfisher87** [comment]
I'm seeing the same error with `sass --version` when installing `1.58` from conda-forge:

```bash
$ sass --version
Unexpected exception:
Invalid argument (uri): Value must be a String or a Uri: null


package:path/src/context.dart 1119                                                      _parseUri
package:path/src/context.dart 1040                                                      Context.fromUri
package:path/path.dart 435                                                              fromUri
../conda/feedstock_root/build_artifacts/dart-sass_1678130103876/work/bin/sass.dart 135  _loadVersion
../conda/feedstock_root/build_artifacts/dart-sass_1678130103876/work/bin/sass.dart 45   main

$ echo $?
255
```

`1.32` doesn't return a version, but also returns 0

```bash
$ sass --version

$ echo $?
0
```

I think this issue may need to be addressed in the `dart-sass-feedstock`?

**Comment by mfisher87** [comment]
For the heck of it, I tried building with `1.32.8` and `quarto check` doesn't like the fact that `dart-sass` returns an empty string for a version :)

```
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
ERROR: TypeError: Invalid Version: 

```

I guess this check wasn't present in Quarto 1.2.x? :shrug:

**Comment by mfisher87** [comment]
@msarahan another PR for you on another repo you maintain :) Opened https://github.com/conda-forge/dart-sass-feedstock/pull/11 to deal with the dart-sass version error!

File: recipe/0001-add-external-installer-command-for-packagers-like-co.patch [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
++import { info } from "log/mod.ts";
++
++import { Configuration } from "../common/config.ts";
++
++export async function makeInstallerExternal(
++  configuration: Configuration,
++) {
++  info("Copying Quarto files...");
++
++  // Ultimate control is setting this env var. Fallback is a folder within QUARTO_DIST_PATH.
```
Comment (mfisher87) [inline recipe/0001-add-external-installer-command-for-packagers-like-co.patch:71]: >Ultimate control is setting this env var

I don't see an envvar corresponding with this comment! Am I missing something?

File: recipe/build.sh [hunk 2]
```diff
@@ -16,19 +16,17 @@ export QUARTO_DENO_DOM=$DENO_DOM_PLUGIN
...
 
-source package/src/set_package_paths.sh
 source configuration
+source package/src/set_package_paths.sh
 
 # This is patched in for conda. This is otherwise set as a constant in `configuration`
 export QUARTO_VERSION=$PKG_VERSION
 
 bash configure.sh
 bash package/src/quarto-bld prepare-dist
```
Comment (mfisher87) [inline recipe/build.sh:28]: I expected we might be deleting this line.  Not so?

**Comment by mfisher87** [comment]
Posted a couple questions, but I think I'll merge this now to reduce the number of PRs in flight :) Thanks Mike!

File: recipe/0001-add-external-installer-command-for-packagers-like-co.patch (hunk 1 above)
Comment (msarahan) [reply recipe/0001-add-external-installer-command-for-packagers-like-co.patch:71]: No, you're just hitting my usual crappy habit of not cleaning up comments after cleaning up code. Thanks for pointing this out.

File: recipe/build.sh (hunk 2 above)
Comment (msarahan) [reply recipe/build.sh:28]: prepare-dist is doing the building part, like compiling the typescript stuff to javascript or whatever. It's still necessary.

Comment (mfisher87) [reply recipe/build.sh:28]: Thanks for answering my questions :)

**Pull request merged**